tests/flow/test_list.py

```diff
@@ -143,3 +143,633 @@ def test03_null_list_function_inputs(self):
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Float", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList(7)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Integer", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList('abc')")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was String", str(e))
+
+        try:
+            query = """CREATE (a:X)"""
+            redis_graph.query(query)
+            query = """MATCH (n) RETURN toBooleanList(n)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Node", str(e))
+            query = """MATCH (a:X) DELETE a"""
+            redis_graph.query(query)
+
+        try:
+            query = """CREATE (a:X), (b:Y)"""
+            redis_graph.query(query)
+            query = """MATCH (a:X), (b:Y) create (a)-[r:R]->(b)"""
+            redis_graph.query(query)
+            query = """MATCH ()-[e]->() RETURN toBooleanList(e)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Edge", str(e))
+            query = """MATCH (a:X),(b:Y) DELETE a, b"""
+            redis_graph.query(query)
+
+        try:
+            query = """CREATE (a:X), (b:Y)"""
+            redis_graph.query(query)
+            query = """MATCH (a:X), (b:Y) CREATE (a)-[r:R]->(b)"""
+            redis_graph.query(query)
+            query = """MATCH p=()-[]->() RETURN toBooleanList(p)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Path", str(e))
+            query = """MATCH (a:X),(b:Y) DELETE a, b"""
+            redis_graph.query(query)
+
+        # List of node input should return list of NULL
+        query = """CREATE (a:X)"""
+        redis_graph.query(query)
+        query = """MATCH (n) RETURN toBooleanList([n])"""
+        expected_result = [[None]]
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+        query = """MATCH (a:X) DELETE a"""
+        redis_graph.query(query)
+
+        # # List of edge input should return list of NULL
+        query = """CREATE (a:X), (b:Y)"""
+        redis_graph.query(query)
+        query = """MATCH (a:X), (b:Y) CREATE (a)-[e:R]->(b)"""
+        redis_graph.query(query)
+        query = """MATCH ()-[e]->() RETURN toBooleanList([e])"""
```

Comment:
consider compacting 3 queries into a single one

Reply:
Done. Query compacted in [b36f30d](https://github.com/RedisGraph/RedisGraph/pull/2605/commits/b36f30dfbf0af4709ba9aa7c5ba4b01be17b10fd)